src/libmongoc/tests/test-mongoc-bulk.c

```diff
@@ -5011,6 +5011,35 @@ test_bulk_write_multiple_errors (void *unused)
    mongoc_client_destroy (client);
 }
 
+// test_bulk_write_set_client_after_operation is a regression test for
+// CDRIVER-4665.
+static void
+test_bulk_write_set_client_after_operation (void)
```

Comment:
Thanks for adding a regression test. I was a bit pressed for time when I created [CDRIVER-4665]https://jira.mongodb.org/browse/CDRIVER-4665) earlier an intentionally omitted that.